src/api.ts

```diff
@@ -142,3 +142,90 @@ export async function createAccessPass(
   );
   return response.data;
 }
+
+export type FieldValue = boolean | string | number;
+
+export interface TestResults {
+  details: { [key: string]: FieldValue }; // TODO: proper typing based on test type
+  testerUserId: string;
+  creationTime: string;
+}
+
+export interface Test {
+  id: string;
+  userId: string;
+  testTypeId: string;
+  creationTime: string;
+  results?: TestResults;
+}
+
+export async function fetchTest(testId: string, options: AuthenticatedHttpOptions): Promise<Test> {
+  const response = await authenticated(options.token).get(`/api/v1/tests/${testId}`, {
+    cancelToken: options.cancelToken,
+  });
+  return response.data;
+}
+
+export async function fetchTests(
+  userId: string,
+  options: AuthenticatedHttpOptions,
+): Promise<Test[]> {
+  const response = await authenticated(options.token).get(`/api/v1/users/${userId}/tests`, {
+    cancelToken: options.cancelToken,
+  });
+  return response.data;
+}
+
+export type FieldType = 'boolean' | 'string' | 'number' | 'integer' | 'null';
+
+export interface FieldSchema {
+  type: FieldType;
+  title: string;
+  description?: string;
+}
+
+// TODO: we are currently hardcoding only simple object -> property json schemas.
+// Should support larger range in the future.
+export interface ObjectSchema {
+  type: 'object';
+  title: string;
+  properties: { [key: string]: FieldSchema };
```

Comment:
Could support description here as well, might provide the ability to give a bit more context to the result format.